### PR TITLE
#mission_nps_mobile_40: Fix hardware back button redirections in android app

### PR DESCRIPTION
**Pull request by yksolanki9**
Notion Doc - https://www.notion.so/fyleuniverse/Fix-Android-Hardware-Back-button-97f533f2a020485db9239bcbf1eda7cc

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -2441,6 +2444,10 @@ export class AddEditExpensePage implements OnInit {
   }
 
   ionViewWillEnter() {
+    this.hardwareBackButtonAction = this.platform.backButton.subscribeWithPriority(20, () => {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2447]: The priority can you convert into a injected value? That will make debugging easier and make sure we assign the proper priority every place

Comment (Julias0) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2447]: The priority can have different levels - base, medium, high, absolute

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comments

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 1 above)
Comment (yksolanki9) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts]: Added enum as discussed

**Review by Dimple16 — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

**Pull request merged**